gtsam/discrete/DecisionTree.h

```diff
@@ -108,19 +122,20 @@ namespace gtsam {
     template<typename It, typename ValueIt>
     NodePtr create(It begin, It end, ValueIt beginY, ValueIt endY) const;
 
-    /** Convert to a different type */
-    template<typename M, typename X> NodePtr
-    convert(const typename DecisionTree<M, X>::NodePtr& f, const std::map<M,
-        L>& map, std::function<Y(const X&)> op);
+    /// Convert from a DecisionTree<M, X>.
+    template <typename M, typename X>
+    NodePtr convertFrom(const typename DecisionTree<M, X>::NodePtr& f,
+                        std::function<L(const M&)> L_of_M,
+                        std::function<Y(const X&)> Y_of_X) const;
```

Comment:
Also, does it make sense to name these `M_to_L` and `X_to_Y`? That way it reads `functor to convert from M to L`. I can make this change if you agree.

Reply:
No, I really dislike "to" names, as you get

```
to_value = From2To(from_value)
```

instead of the natural flow

```
to_value = ToOfFrom(from_value)
```

That's why matrix convention is the way it is